src/TorchSharp/Tensor/Tensor.Factories.cs

```diff
@@ -1369,10 +1369,18 @@ public static Tensor tensor(IList<bool> rawArray, long[] dimensions, torch.Scala
         /// <summary>
         /// Create a tensor from an array of values, shaping it based on the shape passed in.
         /// </summary>
-        public static Tensor tensor(bool[] dataArray, long[] dimensions, torch.ScalarType? dtype = null, torch.Device? device = null, bool requiresGrad = false)
+        public static Tensor tensor(bool[] rawArray, long[]? dimensions = null, torch.ScalarType? dtype = null, torch.Device? device = null, bool requiresGrad = false)
+        {
+            if (dimensions is null)
+                dimensions = new long[] { rawArray.LongLength };
+
+            var dataHandle = GCHandle.Alloc(rawArray, GCHandleType.Pinned);
+            return _tensor_generic(dataHandle, dimensions, (sbyte)ScalarType.Bool, dtype, device, requiresGrad);
+        }
+
+        private static Tensor _tensor_generic(GCHandle dataHandle, long[] dimensions, sbyte origType, ScalarType? dtype, Device? device, bool requiresGrad)
```

Comment:
>_tensor_generic [](http://example.com/codeflow?start=30&length=15)

why this method not taking the rawArray as object and does the allocation inside? this will not require all caller to do the allocation manually before calling it.